File: src/main/java/com/apsi/service/dto/TransactionDTO.java
package com.apsi.service.dto;

import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class TransactionDTO {

    private Long id;

    private String side;

    private Float insertedAmount;

    private String currency1;

    private Float calculatedAmount;

    private String currency2;

    private Float exchange;

    private String billAccountK;

    private String billAccountS;

    private Integer userId;

    private LocalDateTime inclusionDate;
}

File: src/main/webapp/app/menu/dto/transaction-dto.ts
export class TransactionDTO {

    id: number;
    side: string;
    insertedAmount: number;
    currency1: string;
    calculatedAmount: number;
    currency2: string;
    exchange: number;
    billAccountK: string;
    billAccountS: string;
    userId: number;
    inclusionDate: string;

    constructor(side: string, insertedAmount: number, currency1: string, calculatedAmount: number, currency2: string,
                exchange: number, billAccountK: string, billAccountS: string, userId: number) {
        this.side = side;
        this.insertedAmount = insertedAmount;
        this.currency1 = currency1;
        this.calculatedAmount = calculatedAmount;
        this.currency2 = currency2;
        this.exchange = exchange;
        this.billAccountK = billAccountK;
        this.billAccountS = billAccountS;
        this.userId = userId;
    }
}
